text/0000-new-test-waiters.md

```diff
@@ -85,21 +85,114 @@ export default class Friendz extends Component {
+
+To annotate the example provided above:
+
+```js
+import Component from '@ember/component';
+import { buildWaiter } from 'ember-test-waiters';
+
+// Creates a test waiter with the name 'friend-waiter' that
+// is usable by all instances of the `Friendz` component.
+let waiter = buildWaiter('friend-waiter');
+
+export default class Friendz extends Component {
+  didInsertElement() {
+    // Alerts the test waiter system that an async operation has started,
+    // storing the resulting unique token to be used to notify the test
+    // waiter system that the operation has ended.
+    let token = waiter.beginAsync();
+
+    someAsyncWork()
+      .then(() => {
+        //... some work
+      })
+      .finally(() => {
+        // Notifies the test waiter system that
+        // this unique async operation has ended.
+        waiter.endAsync(token);
+      });
+  }
+}
+```
+
+### `waitForPromise`
+
+The `waitForPromise` utility provides a convenience wrapper around the `TestWaiter` class for use with promises. It ensures the `endAsync` call is invoked in the `finally` of the configured promise.
 
-This new test waiters system has been through multiple iterations of refinement, and is in use and integrated with the test isolation validation system. Furthermore, it is backwards compatible with the old test waiters system, allowing applications and addons to gradually migrate to using the new system.
+This new test waiters system has been through multiple iterations of refinement, and is in use and integrated with the test isolation validation system.
 
-## Deprecating the old test waiters
+## Rename of `ember-test-waiters` to `@ember/test-waiters`
 
-The old test waiters system will be deprecated in its own deprecation RFC.
+We should consider renaming the `ember-test-waiters` repository to `@ember/test-waiters`, which would relocate it to the Ember org and scope it more explicitly.
+
+## Compatibility
+
+The `ember-test-waiters` addon is backwards compatible with the old test waiters system, allowing applications and addons to gradually migrate to using the new system.
+
+Specifically:
+
+- `registerWaiter` continues to work, and is not deprecated
+- addons using `ember-test-waiters` work _even if consumed in applications that do not use a new enough `ember-test-helpers` version_ (they won't get additional output via test isolation validation such as test waiter names or stack traces)
+
+The old test waiters system ultimately should be deprecated in its own deprecation RFC.
 
 # How We Teach This
 
 This new test waiters system should be included in the Ember guide's testing section. Information and examples should be provided to allow users to correctly author asynchronous code that can be correctly managed by the testing system.
 
+Specifically, calling this out in a separate section will allow readers to
```

Comment:
Seems like this is cut off?

Reply:
Correct. I left the rest in my brain. Fixed now.